datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -132,49 +148,45 @@ class except the :py:meth:`check` method but sometimes it might be useful for a
                 configuration file (a list is used to keep backward compatibility with
                 older versions of the Agent).
         """
-        self.metrics = defaultdict(list)
-        self.check_id = ''
-        self.instances = kwargs.get('instances', [])
-        self.name = kwargs.get('name', '')
-        self.init_config = kwargs.get('init_config', {})
-        self.agentConfig = kwargs.get('agentConfig', {})
-        self.warnings = []
+        name = kwargs.get('name', '')
```

Comment:
Might be helpful to add a comment that these assignments are being done to ease type checking.

Reply:
Good idea — added in b01c94e61.